Gems/Atom/RHI/DX12/Code/Source/RHI/DescriptorContext.h

```diff
@@ -147,6 +162,25 @@ namespace AZ
             DescriptorHandle m_nullSamplerDescriptor;
 
             RHI::ConstPtr<PlatformLimitsDescriptor> m_platformLimitsDescriptor;
+
+            // Use 2 heaps below in order to ping-pong between shader visible CbvSrvUav heap when one of them fragments and run out of memory.
```

Comment:
I think we should consider a more stable abstraction which is to create a new heap when we're out of memory or have fragmented beyond our limits (so that the number of heaps is actually variable instead of fixed to 2)

Reply:
In this specific example you shouldnt need more than two heaps otherwise you are just wasting memory. Remember this only for gpu visible heap and you can only bind one in a frame. I could modify it so that we dont allocate both the heaps at start and only allocate the second one once fragmentation occurs and then after that we wouldnt need anymore (Let me know if you want that). If for some reason we still run out handles (i.e during compaction phase) the solution here would be to increase the number of handles allowed within setting registry. At every point I have an assert with appropriate message in place that directs the developer with a solution